.github/workflows/ghcr_runtime.yml

```diff
@@ -72,77 +74,17 @@ jobs:
-            # Capture the last tag to use in the artifact name
-            last_tag=$(cat tags.txt | awk '{print $NF}')
-          else
-            echo 'No Dockerfile detected which means an exact image is already built. Pulling the image and saving it to a tar file...'
-            source containers/runtime/config.sh
-            echo "$DOCKER_IMAGE_HASH_TAG $DOCKER_IMAGE_TAG" >> tags.txt
-            export last_tag=$DOCKER_IMAGE_TAG
-            echo "Pulling image $DOCKER_REGISTRY/$DOCKER_ORG/$DOCKER_IMAGE:$DOCKER_IMAGE_HASH_TAG to /tmp/${{ matrix.image }}_${last_tag}_${{ matrix.platform }}.tar"
-            docker pull $DOCKER_REGISTRY/$DOCKER_ORG/$DOCKER_IMAGE:$DOCKER_IMAGE_HASH_TAG
-            docker save $DOCKER_REGISTRY/$DOCKER_ORG/$DOCKER_IMAGE:$DOCKER_IMAGE_HASH_TAG -o /tmp/${{ matrix.image }}_${last_tag}_${{ matrix.platform }}.tar
-          fi
-          echo "last_tag=${last_tag}" >> $GITHUB_OUTPUT
-      - name: Capture tags
-        id: capture-tags
-        run: |
-          tags=$(cat tags.txt)
-          echo "tags=$tags"
-          echo "tags=$tags" >> $GITHUB_OUTPUT
-      - name: Upload Docker image as artifact
-        uses: actions/upload-artifact@v4
-        with:
-          name: ${{ matrix.image }}_${{ steps.build.outputs.last_tag }}_${{ matrix.platform }}
-          path: /tmp/${{ matrix.image }}_${{ steps.build.outputs.last_tag }}_${{ matrix.platform }}.tar
-          retention-days: 14
-      - name: Capture last tag
-        id: capture-last-tag
-        run: |
-          last_tag=$(cat tags.txt | awk '{print $NF}')
-          echo "$last_tag" > /tmp/last-tag-${{ matrix.image }}-${{ matrix.platform }}-${{ steps.build.outputs.last_tag }}.txt
-          echo "Saved last tag to /tmp/last-tag-${{ matrix.image }}-${{ matrix.platform }}-${{ steps.build.outputs.last_tag }}.txt"
-      - name: Upload last tag as artifact
-        uses: actions/upload-artifact@v4
-        with:
-          name: last-tag-${{ matrix.image }}-${{ matrix.platform }}-${{ steps.build.outputs.last_tag }}
-          path: /tmp/last-tag-${{ matrix.image }}-${{ matrix.platform }}-${{ steps.build.outputs.last_tag }}.txt
-          retention-days: 1
-
-  prepare_test_image_tags:
-    name: Prepare Test Images Tags
-    needs: ghcr_build_runtime
-    runs-on: ubuntu-latest
-    outputs:
-      test_image_tags: ${{ steps.set-matrix.outputs.test_image_tags }}
-    steps:
-      - name: Download last tags
-        uses: actions/download-artifact@v4
-        with:
-          pattern: last-tag-*
-          path: /tmp/
-          merge-multiple: true
-      - name: Set up test matrix
-        id: set-matrix
-        run: |
-          matrix=$(cat /tmp/last-tag-*.txt | sort -u | jq -R -s -c 'split("\n") | map(select(length > 0))')
-          echo "test_image_tags=$matrix" >> $GITHUB_OUTPUT
-          echo "Generated test_image_tags: $matrix"
+          suffix=$(echo "${{ matrix.base_image }}" | cut -d ':' -f 1 | cut -d '/' -f 1)
```

Comment:
We might need better approach than this going forward, but probably good for now

Reply:
yeah agreed